metricbeat/mb/registry.go

```diff
@@ -243,18 +263,29 @@ func (r *Register) DefaultMetricSets(module string) ([]string, error) {
 
 	module = strings.ToLower(module)
 
+	var defaults []string
 	metricSets, exists := r.metricSets[module]
-	if !exists {
-		return nil, fmt.Errorf("module '%s' not found", module)
+	if exists {
+		for _, reg := range metricSets {
+			if reg.IsDefault {
+				defaults = append(defaults, reg.Name)
+			}
+		}
 	}
 
-	var defaults []string
-	for _, reg := range metricSets {
-		if reg.IsDefault {
-			defaults = append(defaults, reg.Name)
+	if source := r.secondarySource; source != nil && source.HasModule(module) {
+		exists = true
+		sourceDefaults, err := source.DefaultMetricSets(module)
+		if err != nil {
+			logp.Error(err)
```

Comment:
should we return here?

Reply:
At the moment this method only fails if a module doesn't exist or if it doesn't have default metricsets, I didn't want to change this behaviour adding a new error condition. But yeah, maybe it is better to fail here more explicitly, this would mean that a module is broken.

Reply:
I have second thoughts on this, for coherence we should also return the error in `MetricSets()` and for mixed modules this can mean that some kind of problem on a secondary source will prevent the use of correct registered metricsets. I am going to keep this only logging the error, we can revisit the decission later as it is keeping current behaviour. I will add more context here in any case.